### PR TITLE
Crud extension

### PR DESCRIPTION
**Pull request by Jatin-tec**
### Description
This pull request addresses [Issue #55](https://github.com/opensearch-project/opensearch-sdk-py/issues/55) by implementing an example CRUD extension in the samples directory. The extension demonstrating basic CRUD operations on an index will and how to handle CRUD operations using OpenSearch SDK for Python with documentation. This addition serves as a reference for developers looking to create similar extensions, showcasing best practices and usage patterns.

### Issues Resolved
Closes [#55](https://github.com/opensearch-project/opensearch-sdk-py/issues/55).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: samples/crud_extension/README.md
Comment (dbwiddis) [inline samples/crud_extension/README.md:56]: The sample doc here and in the GET doesn't include the rating in the PUT.  Is that intentional?  

These instructions could use a bit more narrative before the commands describing what we are doing.

File: samples/crud_extension/__init__.py
Comment (dbwiddis) [inline samples/crud_extension/__init__.py:8]: This file (and others) are missing a final CR before EOF.  CI shows other formatting issues.

We have several formatters and linters that fix all this for you; see https://github.com/opensearch-project/opensearch-sdk-py/blob/main/DEVELOPER_GUIDE.md#code-linting

File: samples/crud_extension/crud_handler.py
Comment (dbwiddis) [inline samples/crud_extension/crud_handler.py:68]: (Style preference) Would prefer you create a consumed_params list and then append to it at the time you access the param.  Not a big deal as it'll all get fixed by #49 eventually but if you're making other changes it would make it more readable here.  Same comment for `id` in other classes.

File: samples/crud_extension/crud_handler.py
Comment (dbwiddis) [inline samples/crud_extension/crud_handler.py:104]: OK for now to just have the same path and use conditionals based on the method.  Eventually when #45 is implemented I think we should update this sample to use some params inline in the path.

File: samples/crud_extension/crud_handler.py
Comment (dbwiddis) [inline samples/crud_extension/crud_handler.py:74]: Note here you are "consuming" the content and should in theory set that boolean to true to return with the response later.

Presently the only OpenSearch failure occurs when a required (named wildcard) parameter is omitted, and you haven't used that feature (because it has a bug!) so your code works... and it will eventually get solved by #49 anyway.

**Review by dbwiddis — COMMENTED** [review]
Fantastic work, @Jatin-tec !  Thanks for your contribution!

I left a few comments, but most of them need to wait until we get #45 and #49 fixed, and we can update the sample then.

File: samples/crud_extension/__init__.py
Comment (Jatin-tec) [reply samples/crud_extension/__init__.py:8]: Thank you for the guidance. I've implemented the suggested changes.

File: samples/crud_extension/README.md
Comment (Jatin-tec) [reply samples/crud_extension/README.md:56]: >  The sample doc here and in the GET doesn't include the rating in the PUT. Is that intentional?

Yes was trying to show how one can update a document.

> These instructions could use a bit more narrative before the commands describing what we are doing.

I'll be on it.

File: samples/crud_extension/__init__.py
Comment (dbwiddis) [reply samples/crud_extension/__init__.py:8]: > Thank you for the guidance. I've implemented the suggested changes.

Haven't seen you push a new commit here yet?

**Comment by dbwiddis** [comment]
Hey @Jatin-tec thanks for updating the formatting.

Linting is failing on something that I'm not sure how to fix, maybe @dblock can clarify:
```
samples/crud_extension/crud.py:12: error: Module "crud_extension" has no attribute "CRUDExtension"  [attr-defined]
```
Also the DCO check is failing. Since you've signed at least one and we'll squash the commits we can work with that but if you can easily retroactively sign it'll make the CI happier.

**Comment by dblock** [comment]
@Jatin-tec lmk if you can't figure out the lint problem and I can take a look

**Comment by Jatin-tec** [comment]
I have incorporated all the suggested changes hope this will resolve the issue and thank you @dbwiddis  and @dblock  for your fantastic support. Your encouragement made my first contribution experience truly amazing and I learned a lot. Thank you!

**Comment by dbwiddis** [comment]
Hey @Jatin-tec thanks again for your work and patience through the process.  I think we can take it from here.

I rebased your code with the main branch since GitHub couldn't figure out a merge conflict in the poetry lock file.  

CI is still failing on mypy error:
```
$ poetry run mypy .          
samples/crud_extension/crud.py:12: error: Module "crud_extension" has no attribute "CRUDExtension"  [attr-defined]
Found 1 error in 1 file (checked 129 source files)
```
I've been looking into this and trying to resolve it.  I think the problem is that the file name `crud_extension.py` matches the directory/module name `crud_extension`.  Evidence:
 - If I delete the line `from crud_extension import CRUDExtension` and recreate it with a quick fix I get `from samples.crud_extension.crud_extension import CRUDExtension` which results in a different mypy error "Source file found twice under different module names: "crud_extension.crud_extension" and "samples.crud_extension.crud_extension""
 - Deleting the prefix `samples` from makes mypy happy: `from crud_extension.crud_extension import CRUDExtension`
 - I'd commit that change, but this still seems to be an extra step not required by the hello extension.

So is there a better fix here?

**Comment by Jatin-tec** [comment]
> Hey @Jatin-tec thanks again for your work and patience through the process. I think we can take it from here.
> 
> I rebased your code with the main branch since GitHub couldn't figure out a merge conflict in the poetry lock file.
> 
> CI is still failing on mypy error:
> 
> ```
> $ poetry run mypy .          
> samples/crud_extension/crud.py:12: error: Module "crud_extension" has no attribute "CRUDExtension"  [attr-defined]
> Found 1 error in 1 file (checked 129 source files)
> ```
> 
> I've been looking into this and trying to resolve it. I think the problem is that the file name `crud_extension.py` matches the directory/module name `crud_extension`. Evidence:
> 
>     * If I delete the line `from crud_extension import CRUDExtension` and recreate it with a quick fix I get `from samples.crud_extension.crud_extension import CRUDExtension` which results in a different mypy error "Source file found twice under different module names: "crud_extension.crud_extension" and "samples.crud_extension.crud_extension""
> 
>     * Deleting the prefix `samples` makes mypy happy: `from crud_extension.crud_extension import CRUDExtension`
> 
>     * I'd commit that change, but this still seems to be an extra step not required by the hello extension.
> 
> 
> So is there a better fix here?

I can restructure the module layout, I think that should fix this.

**Comment by dbwiddis** [comment]
> I can restructure the module layout, I think that should fix this.

I suspect simply renaming `samples/crud_extension` to `samples/crud` would be sufficient.

Also if you're still iterating, I've handled the consumed content stuff in #76, if you'd like to integrate those changes in your sample I can merge that PR and you can rebase.   Otherwise I'm holding off on merging that and can make changes to your code here after we get this merged.

**Review by dbwiddis — APPROVED** [review]

**Comment by dbwiddis** [comment]
Thanks for this contribution, @Jatin-tec!  Hope you stick around and do more!

**Pull request merged**
